src/backend/renderer/element/surface.rs

```diff
@@ -266,33 +270,51 @@ where
         surface: &wl_surface::WlSurface,
         state: &RendererSurfaceStateUserData,
         location: impl Into<Point<f64, Physical>>,
+        scale: impl Into<Scale<f64>>,
     ) -> Option<Self> {
         let location = location.into();
         let state = state.borrow();
 
         let view = state.view()?;
         let buffer = state.buffer()?;
+        let buffer_type = crate::backend::renderer::buffer_type(buffer)?;
         let id = Id::from_wayland_resource(surface);
 
-        let texture = state.texture::<R>(renderer.id())?;
+        if matches!(buffer_type, crate::backend::renderer::BufferType::SinglePixel) {
+            let spb = crate::wayland::single_pixel_buffer::get_single_pixel_buffer(buffer).ok()?;
+            let rgba = spb.rgba8888();
```

Comment:
So we truncate it to rgba8888, just to convert it to floats afterwards..

Can we add a `rgba16f` variant please?

Reply:
yeah, that is kinda pointless. any reason why to use half float? After reading the spec (and looking at what swaybg sends) the values range from 0 - u32::MAX. So dividing by `u32::MAX as f32` (or just `0xffffffff`) should give us `f32` per component?